client/team_organization_assignment.go

```diff
@@ -0,0 +1,59 @@
+package client
+
+type AssignOrganizationRoleToTeamPayload struct {
+	TeamId string `json:"teamId"`
+	Role   string `json:"role" tfschema:"role_id"`
+}
+
+type OrganizationRoleTeamAssignment struct {
+	TeamId string `json:"teamId"`
+	Role   string `json:"role" tfschema:"role_id"`
+	Id     string `json:"id"`
+}
+
+func (client *ApiClient) AssignOrganizationRoleToTeam(payload *AssignOrganizationRoleToTeamPayload) (*OrganizationRoleTeamAssignment, error) {
+	var result OrganizationRoleTeamAssignment
+
+	organizationId, err := client.OrganizationId()
```

Comment:
you extract `organizationId` from the client? I guess that's for security reasons, but wouldn't it be clearer that the user will pass the `organizationId` and we'll verify it is equal to the client's `organizationId`?

Reply:
I don't know. I suggest you discuss with the team.
In general every where else in the provider the organization id is implicit (no need to enter it).